src/world.js

```diff
@@ -71,9 +72,28 @@ class World extends EventEmitter {
       }
       const loaded = chunk != null
       if (!loaded && this.chunkGenerator) {
-        chunk = this.chunkGenerator(chunkX, chunkZ)
+        if (this.loadingColumns.includes(key)) {
+          var abstract = this
+          return await new Promise((resolve, reject) => {
+            var iv = setInterval(() => {
+              if (!abstract.loadingColumns.includes(key)) {
+                clearInterval(iv)
+                resolve(abstract.columns[key])
+              }
+            }, 1)
```

Comment:
This setInterval is called too often, it can slow down other stuff (especially since you potentially create one setInterval for each call to getColumn). In general, active pooling is a bad practice.

What you could do instead is store a promise in loadingColumns and return it. That way you only have 1 promise per column and no setInterval.

Something like : (not tested)
```js
if (!loaded && this.chunkGenerator) {
  if (this.loadingColumns[key]) return this.loadingColumns[key]
  chunk = this.chunkGenerator(chunkX, chunkZ) // should always return a promise
  this.loadingColumns[key] = chunk.then((chunk) => {
    delete this.loadingColumns[key]
    if (chunk != null) { await this.setColumn(chunkX, chunkZ, chunk, !loaded) }
    return this.columns[key]
  })
  return this.loadingColumns[key]
}
```